File: python/length_of_longest_substring.py
def length_of_longest_substring(s: str) -> int:
    seen = set()
    left = 0
    sum = 0
    for right in range(len(s)):
        while s[right] in seen:
            seen.remove(s[left])
            left += 1
        seen.add(s[right])
        if right - left + 1 > sum:
            sum = right - left + 1
    return sum

File: Sliding Window/js/lengthOfLongestSubstring.js
const lengthOfLongestSubstring = function (s) {
  let left = 0;
  let sum = 0;
  let seen = new Set();

  for (let right = 0; right < s.length; right++) {
    while (seen.has(s[right])) {
      seen.delete(s[left]);
      left++;
    }
    seen.add(s[right]);

    if (right - left + 1 > sum) {
      sum = right - left + 1;
    }
  }
  return sum;
};
